wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -891,11 +891,19 @@ public final class SessionManager: NSObject, SessionManagerType {
     private func switchAnalyticsUser() {
         guard
             let analyticsManager,
-            let activeUserSession
+            let activeUserSession,
+            let userProfile = getUserAnalyticsProfileForActiveUserSession()
         else {
             return
         }
 
+        let analyticsSession = analyticsManager.switchUser(userProfile)
+
+        activeUserSession.analyticsSession = analyticsSession
+    }
+
+    func getUserAnalyticsProfileForActiveUserSession() -> AnalyticsUserProfile? {
```

Comment:
I suggest to make this an extension on the user session, or a function that accepts the user session as an argument. This would avoid unwrapping the user session twice.

Reply:
I decided to go with the function route: https://github.com/wireapp/wire-ios/commit/bec25454f8ca7ddc9ac4525626707f404f387658